api/src/main/java/jakarta/enterprise/concurrent/CronTrigger.java

```diff
@@ -0,0 +1,665 @@
+    private int[] seconds = ZERO;
+    private final ZoneId zone;
+
+    /**
+     * Constructor that accepts a cron expression.
+     *
+     * @param cron cron expression.
+     * @param zone timezone ID to use for {@link java.time.ZonedDateTime} that is supplied to
+     *        {@link #getNextRunTime(LastExecution, ZonedDateTime) getNextRunTime} and
+     *        {@link #skipRun(LastExecution, ZonedDateTime) skipRun} methods.
+     *        Null indicates to use the system default.
+     */
+    public CronTrigger(String cron, ZoneId zone) {
+        this(zone);
+        String[] c = cron.split(" ");
+        if (c.length == 5)
+            minutes(c[0]).hours(c[1]).daysOfMonth(c[2]).months(c[3]).daysOfWeek(c[4]);
+        else if (c.length == 6)
+            seconds(c[0]).minutes(c[1]).hours(c[2]).daysOfMonth(c[3]).months(c[4]).daysOfWeek(c[5]);
+        else
+            throw new IllegalArgumentException(cron);
+    }
+
+    /**
+     * Constructor for the fluent configuration pattern.
+     * Seconds, minutes, and hours default to 0. The remaining fields default to *.
+     *
+     * @param zone timezone ID to use for {@link java.time.ZonedDateTime} that is supplied to
+     *        {@link #getNextRunTime(LastExecution, ZonedDateTime) getNextRunTime} and
+     *        {@link #skipRun(LastExecution, ZonedDateTime) skipRun} methods.
+     *        Null indicates to use the system default.
+     */
+    public CronTrigger(ZoneId zone) {
+        this.zone = zone == null ? ZoneId.systemDefault() : zone;
+    }
+
+    /**
+     * Using the cron schedule, and based on the end of the most recent execution
+     * (or absent that, the initial scheduling time), retrieve the next time
+     * that the task should run after.
+     *
+     * @param lastExecutionInfo information about the last execution of the task. 
+     *                          This value will be null if the task has not yet run.
+     * @param taskScheduledTime the date/time at which the
+     *                          {@code ManagedScheduledExecutorService.schedule}
+     *                          method was invoked to schedule the task.
+     * @return the date/time after which the next execution of the task should start.
+     * @throws DateTimeException if a next time cannot be determined from the cron expression.
+     */
+    public ZonedDateTime getNextRunTime(LastExecution lastExecutionInfo, ZonedDateTime taskScheduledTime) {
+        return next(lastExecutionInfo == null ? taskScheduledTime : lastExecutionInfo.getRunEnd(zone));
+    }
+
+    /**
+     * Returns the timezone to use for
+     * {@link java.time.ZonedDateTime ZonedDateTime} that is supplied to the
+     * {@link #getNextRunTime(LastExecution, java.time.ZonedDateTime) getNextRunTime} and
+     * {@link #skipRun(LastExecution, java.time.ZonedDateTime) skipRun} methods.
+     *
+     * @return timezone to use for operations on this trigger.
+     */
+    public final ZoneId getZoneId() {
```

Comment:
Does this need @Override

Reply:
Good catch, I added commit https://github.com/eclipse-ee4j/concurrency-api/pull/144/commits/0b8a5dc60e57960de0b610448250079ae8fe7b67 to include `@Override` here and on two other methods where it was missing.